core/modules/memman/Memory.cc

```diff
@@ -0,0 +1,145 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class Header
+#include "memman/Memory.h"
+
+// System Headers
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+/******************************************************************************/
+/*                              f i l e I n f o                               */
+/******************************************************************************/
+  
+MemInfo Memory::fileInfo(std::string const& fPath) {
+
+MemInfo     fInfo;
+struct stat sBuff;
+
+    // Simply issue a stat() to get the size
+    //
+    if (stat(fPath.c_str(), &sBuff)) fInfo._errCode = errno;
+       else {if (sBuff.st_size > 0) fInfo._memSize = sBuff.st_size;
+                else fInfo._errCode = ESPIPE;
+            }
```

Comment:
My eyes :sob: 

Is not it the same as:

    if (stat(fPath.c_str(), &sBuff)) {
        fInfo._errCode = errno;
    } else if (sBuff.st_size > 0) {
        fInfo._memSize = sBuff.st_size;
    } else {
        fInfo._errCode = ESPIPE;
    }